src/deploy.ts

```diff
@@ -342,24 +343,25 @@ export async function deploy(
     workspaceLogin: deployTarget.workspace.login
   });
 
-  // Create the new deploy on the server
-  let deployId: string;
-  try {
-    deployId = await apiClient.postDeploy({projectId: deployTarget.project.id, message});
-  } catch (error) {
-    if (isHttpError(error)) {
-      if (error.statusCode === 404) {
-        throw new CliError(`Project @${deployTarget.workspace.login}/${deployTarget.project.slug} not found.`, {
-          cause: error
-        });
-      } else if (error.statusCode === 403) {
-        throw new CliError(
-          `You don't have permission to deploy to @${deployTarget.workspace.login}/${deployTarget.project.slug}.`,
-          {cause: error}
-        );
+  if (!deployId) {
```

Comment:
This should have an else, and of a deployOd was specified, query the server early to check that it is valid and the user has access. Checking early will enable us to give quicker and more helpful error messages.